src/main/java/org/folio/search/model/rest/response/PagedResult.java

```diff
@@ -1,18 +1,10 @@
 package org.folio.search.model.rest.response;
 
 import java.util.List;
-import lombok.Data;
-import lombok.NoArgsConstructor;
 
-@Data
-@NoArgsConstructor
-public class PagedResult<T> {
+public interface PagedResult<T> {
 
-  private long total;
-  private long totalPages;
+  long getTotalRecords();
 
-  private int pageSize;
-  private int pageNumber;
-
-  private List<T> content;
+  List<T> getInstances();
```

Comment:
Assuming the result could be instances, holdings, or items, maybe consider renaming to `getRecords` to be more generic?